opensrp-chw-malaria/src/main/java/org/smartregister/chw/malaria/presenter/BaseMalariaProfilePresenter.java

```diff
@@ -0,0 +1,60 @@
+package org.smartregister.chw.malaria.presenter;
+
+import android.content.Context;
+import android.support.v4.content.ContextCompat;
+import android.view.View;
+import android.widget.Toast;
+
+import org.smartregister.chw.malaria.contract.MalariaProfileContract;
+import org.smartregister.chw.malaria.domain.MemberObject;
+import org.smartregister.malaria.R;
+
+
+public class BaseMalariaProfilePresenter {
+    protected MalariaProfileContract.View view;
+    protected MemberObject memberObject;
+    protected Context context;
+
+
+    public BaseMalariaProfilePresenter(MalariaProfileContract.View view, MemberObject memberObject) {
+        this.view = view;
+        this.memberObject = memberObject;
+
+    }
+
+    public void attachView(MalariaProfileContract.View view) {
+        this.view = view;
+    }
+
+    public void detachView() {
+        this.view = null;
+    }
+
+    public void fillProfileData(MemberObject memberObject) {
+        if (memberObject != null) {
+            view.setProfileViewWithData();
+        }
+    }
+
+    public void recordMalariaButton(int days_from_malaria_test_date, View view, Context context) {
+        if(days_from_malaria_test_date >= 7 && days_from_malaria_test_date < 10)  {
```

Comment:
is it possible to have a date range outside the ones on the if conditions

Reply:
Do we need that? I could make it more dynamic if needed be.